include/kernel/thread.h

```diff
@@ -280,20 +280,31 @@ static inline bool_t checkBudget(void)
     return false;
 }
 
-/* Everything checkBudget does, but also set the thread
- * state to ThreadState_Restart. To be called from kernel entries
- * where the operation should be restarted once the current thread
- * has budget again.
- */
 
-static inline bool_t checkBudgetRestart(void)
+/* This is called from kernel entries where operation are done only if the
+ * thread still has budget. Otherwise it needs to be restarted first.
+ */
+static inline bool_t checkBudgetWithRestart(void)
```

Comment:
This rename is unnecessary and invalidates the caches for anyone who is deeply familiar with the MCS branch. Please don't rename things unless they are factually wrong.

Reply:
It's not a rename, because the function does more now. It consolidates what the macro `MCS_DO_IF_BUDGET()` did and this was the only place where the remove function was used.